python/kserve/kserve/protocol/grpc/server.py

```diff
@@ -58,8 +63,18 @@ async def start(self, max_workers):
             inference_servicer, self._server
         )
 
-        listen_addr = f"[::]:{self._port}"
-        self._server.add_insecure_port(listen_addr)
+        listen_addr = f'[::]:{self._port}'
+
+        if self._secure_server:
+            server_credentials = grpc_ssl_server_credentials(
```

Comment:
can we have a test for this?

Reply:
I hadn't seen any existing tests for the server start functionality (neither gRPC or REST), so was unsure what the scope of testing was supposed to be here.  can write a few if that would be best, let me know

Reply:
One simple test could be done by creating the certificates and spawn a ssl protect server and try to make any request.

Reply:
Going to test this - as well as the helpers - by creating a folder with a set of dummy SSL certs for testing purposes.  Just noting this for posterity (& will document as well) because they will eventually expire and require refreshing